src/public_api/mod.rs

```diff
@@ -1,20 +1,43 @@
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+
+use crate::decide::{Decide, DecideResponse};
+use crate::errors::Error;
 use crate::event::Event;
 use crate::types::APIResult;
 use crate::Client;
 
+#[async_trait]
 pub trait PublicAPI {
-    fn capture(&self, event: Event) -> APIResult<()>;
-    fn capture_batch(&self, events: Vec<Event>) -> APIResult<()> {
+    async fn capture(&self, event: Event) -> APIResult<()>;
+    async fn capture_batch(&self, events: Vec<Event>) -> APIResult<()> {
         for event in events {
-            self.capture(event)?;
+            self.capture(event).await?;
         }
         Ok(())
     }
+
+    async fn decide(&self, user_id: String) -> Result<HashMap<String, bool>, Error>;
```

Comment:
you can use `APIResult` to indicate that it is a request from the API. 
```suggestion
    async fn decide(&self, user_id: String) -> APIResult<HashMap<String, bool>>;
```

Also you should prefer to use `&str` vs `String` here... and personally I think it's even better to use a `type` to indicate that it is a user_id as done in [types.rs](https://github.com/Storykit/posthog-rs/blob/main/src/types.rs)

The reason for `&str` vs `String` is that users doesn't have to do anything with the parameter when they use the function, they can just write: `client.decide("123482956")`